superset/common/query_object.py

```diff
@@ -214,8 +223,34 @@ def __init__(
 
     @property
     def metric_names(self) -> List[str]:
+        """Return metrics names (labels), coerce adhoc metrics to strings."""
         return get_metric_names(self.metrics)
 
+    @property
+    def column_names(self) -> List[str]:
+        """Return column names (labels). Reserved for future adhoc calculated
+        columns."""
+        return self.columns
+
+    def validate(
+        self, raise_exceptions: Optional[bool] = True
+    ) -> Optional[QueryObjectValidationError]:
+        """Validate query object"""
+        error: Optional[QueryObjectValidationError] = None
+        all_labels = self.metric_names + self.column_names
+        if len(set(all_labels)) < len(all_labels):
+            dup_labels = find_duplicates(all_labels)
+            error = QueryObjectValidationError(
+                _(
+                    "Duplicate column/metric labels: %(labels)s. Please make "
+                    "sure all columns and metrics have a unique label.",
+                    labels=", ".join(map(lambda x: f'"{x}"', dup_labels)),
```

Comment:
nit (might require the `TypeVar` suggestion below):
```suggestion
                    labels=", ".join(dup_labels),
```

Reply:
This is about adding quotes around the labels, e.g. 

```
Duplicate labels:  Population of Canada, Population of USA, Time.
```

v.s.

```
Duplicate labels: "Population of Canada", "Population of USA", "Time".
```

Can `TypeVar` support that?

Reply:
Oh right, I missed that - It's customary to use a list comprehension here, so I'd rather just do 
```python
                    labels=", ".join([f'"{label}"' for label in dup_labels]),
```
to keep it in line with conventions used elsewhere in the codebase.

Reply:
Sure thing! Just one more note, `[...]` is not needed since `.join` accepts an iterator as input:

```
labels=", ".join(f'"{label}"' for label in dup_labels),
```

Reply:
Oh nice, good to know!